python/pyfunc-server/test/test_http.py

```diff
@@ -59,45 +95,32 @@ def infer(self, model_input: ModelInput) -> ModelOutput:
         return sample_model_output
     def postprocess(self, model_output: ModelOutput, request: dict) -> dict:
         return model_output.predictions_dict()
-
-
-class LoadTestModelV3(PyFuncV3Model):
+    
+class SamplePyFuncV3NumpyModel(PyFuncV3Model):
     def preprocess(self, request: dict, **kwargs) -> ModelInput:
-        features_data = request["features"]
-        columns  = list(features_data[0].keys())
-        data = []
-        row_ids = []
-        for feature in features_data:
-            row_ids.append(feature["prediction_id"])
-            row = []
-            for name in feature:
-                row.append(feature[name])
-            data.append(row)
-        model_input = ModelInput(
-            features=Values(
-                columns=columns,
-                data = data
-            ),
-            prediction_ids = row_ids
-        )
-        return model_input
+        return sample_model_input_np
+    def infer(self, model_input: ModelInput) -> ModelOutput:
+        return sample_model_output_np
+    def postprocess(self, model_output: ModelOutput, request: dict) -> dict:
+        return model_output.predictions_dict()
     
+class SamplePyFuncV3DFModel(PyFuncV3Model):
+    def preprocess(self, request: dict, **kwargs) -> ModelInput:
+        return sample_model_input_df
     def infer(self, model_input: ModelInput) -> ModelOutput:
-        predictions = [random.uniform(0,1) for _ in range(len(model_input.prediction_ids))]
-        data = []
-        for p in predictions:
-            data.append([p])
-        
-        return ModelOutput(
-            predictions=Values(
-                columns=["prediction_score"],
-                data = data
-            ),
-            prediction_ids = model_input.prediction_ids
-        )
+        return sample_model_output_df
     def postprocess(self, model_output: ModelOutput, request: dict) -> dict:
         return model_output.predictions_dict()
 
+
+class SamplePyFuncWithModelObservability(PyFuncModel):
+    def infer(self, request, **kwargs) -> ModelOutput:
```

Comment:
I think we should also update the signature of the `infer` method for the `PyFuncModel` to better reflect this.

```python
def infer(self, model_input: Union[dict, ModelInput]) -> Union[dict, ModelOutput]:
```

This will also take us a step closer to unifying the 2 PyFunc types later on. Wdyt?

Reply:
For now I'll leave it as it is, will modify it in the another PR to unifying the pyfuncv3 and pyfunc